### DISCUSSION TITLE
Permanently modifying the CSS for my self-hosted install ...

### DISCUSSION THREAD

**Discussion by ReessKennedy** [outline/outline#4206] [General]
## My preference
I greatly prefer the standard, prominent blue hyperlinks in my documentation and the use of the ":visited" selector in purple to remind me whether I have visited this link instead of Outline's black link. I also prefer it when links are the same font weight as the text. 

## My temporary solution
As a result of this preference, I am using a Chrome extension to insert my own CSS while I am visiting the link to my Outline Wiki install BUT I now would like to insert this little bit of CSS directly into Outline so all my team members see this design by default as well! ... (and don't have to use an extension) ... https://chrome.google.com/webstore/detail/user-javascript-and-css/nbhcbdghjpllgmfilhnhkllmkecfmpld?hl=en

## Seeking a better solution
When using a docker stack to deploy this does anyone have any ideas for how I could do this and still be able to stay on top of updates? ... it would be annoying to have to manually go in and add my own CSS each time. Perhaps these is a custom .SCSS file I could populate with this code and then it would be inserted on build? One idea I had is to possible insert this in Cloudflare as I think there is an option for CSS insertion but this is till still kind of a hack. Optimally I cold customize the Outline build. Of course, it might also be nice if Outline just added an easy place in settings to insert CSS mods!

`
@import url('https://rsms.me/inter/inter.css');
html, p { font-family: 'Inter', sans-serif; }
p a {
    color: rgb(17 73 242);
    text-decoration: underline 1px rgb(17 73 242);
    text-underline-offset: 0.15em;
    font-weight: 400;
}

#root > div > div > div > div.gfQdxL p a:link {
	color: #0000EE !important;
	font-weight: 400 !important;
	text-decoration: underline 1px #0000EE;
}
#root > div > div > div > div.gfQdxL p a:visited {
	color: #551A8B !important;
}
`

<img width="966" alt="Screen Shot 2022-10-01 at 6 03 16 PM" src="https://user-images.githubusercontent.com/3970267/193429987-b90426e0-9807-47ed-b2e1-ab83d288d8b3.png">

**Comment by ReessKennedy** [comment]
Instead of messing with customizing the build I decided to insert CSS for all team members using a Cloudflare app called "Add CSS" ... it's working well. Of course, downside is you'd have to be using Cloudflare and it will break if you move but this is likely easier than maintaining a custom build just for a CSS change. 

This is the app used: 
<img width="285" alt="Screen Shot 2022-10-17 at 12 00 22 PM" src="https://user-images.githubusercontent.com/3970267/196228115-61fc34b4-34f3-431e-b912-3c9f4b69e7da.png">

**Comment by antran22** [comment]
I think that you guys can allow system admin to directly inject CSS into the page, which can be named theme. This can be very helpful with white-labelling without being very difficult to implement.

Obviously this is a power-feature and could break on update very easily. What are the maintainers' thoughts on this?

**Comment by mweth** [comment]
A way of defining classes and applying them with shortcodes may be a temporary work around.

**Comment by praveenpuglia** [comment]
Doing the same with a chrome extension called - Stylebot. :|

**Comment by seanpai96** [comment]
any updates to this feature? It is very hard to distinguish between links and underscored text without clicking on it.
![image](https://github.com/outline/outline/assets/35644330/f60d41b0-8030-4894-be06-6297f55bbbdd)

**Comment by sfakir** [comment]
+1

**Comment by denizbinay** [comment]
+1

**Comment by wforte88** [comment]
+1

**Comment by EduNetEurope** [comment]
@tommoor 
Can we turn this discussion into a feature request?
We would also love to add custom CSS to adapt the visual appearance.

**Comment by mary-loulou** [comment]
Hi @tommoor
Is custom CSS on your radar? There are many requests asking for mainly changing colours and spacing:
- https://github.com/outline/outline/discussions/2362
- https://github.com/outline/outline/discussions/8537
- https://github.com/outline/outline/discussions/7720
- https://github.com/outline/outline/discussions/4307

**Reply by pricard32** [reply]
+1 for custom CSS.
